pkg/delete/loader.go

```diff
@@ -137,99 +120,104 @@ func parseDeleteFileDefinition(ctx *loaderContext, definition persistence.FileDe
-
+func convert(entry interface{}) (pointer.DeletePointer, error) {
 	var parsed persistence.DeleteEntry
-	err := mapstructure.Decode(entry, &parsed)
-	if err != nil {
+	if err := mapstructure.Decode(entry, &parsed); err != nil {
 		return pointer.DeletePointer{}, err
 	}
 
-	if a, known := ctx.knownApis[parsed.Type]; known {
-		p, err := parseAPIEntry(parsed, a)
-		if err != nil {
+	if parsed.Type == "" {
+		return pointer.DeletePointer{}, errors.New("'type' is not supported for this API")
+	}
+	if a, known := api.NewAPIs()[parsed.Type]; known {
+		if err := verifyAPIEntry(parsed, a); err != nil {
 			return pointer.DeletePointer{}, fmt.Errorf("failed to parse entry for API '%s': %w", a.ID, err)
 		}
-		return p, nil
+	} else {
+		if err := verifyCoordinateEntry(parsed); err != nil {
+			return pointer.DeletePointer{}, err
+		}
 	}
 
-	return parseCoordinateEntry(parsed)
-}
-
-func parseAPIEntry(parsed persistence.DeleteEntry, a api.API) (pointer.DeletePointer, error) {
-	if parsed.ConfigName == "" {
-		return pointer.DeletePointer{}, fmt.Errorf("delete entry of API type requiress config 'name' to be defined")
+	dp := pointer.DeletePointer{
+		Project:    parsed.Project,
+		Type:       parsed.Type,
+		Scope:      parsed.Scope,
+		ActionType: parsed.CustomValues["actionType"],
+		Domain:     parsed.CustomValues["domain"],
+		OriginID:   parsed.ObjectId,
 	}
-
-	if parsed.ConfigId != "" {
-		log.Warn("Delete entry '%s' of API type defines config 'id' - only 'name' will be used.")
+	if _, known := api.NewAPIs()[parsed.Type]; known {
+		dp.Identifier = parsed.ConfigName
+	} else {
+		dp.Identifier = parsed.ConfigId
 	}
 
-	if a.ID == api.KeyUserActionsWeb {
-		if parsed.Scope == "" {
-			return pointer.DeletePointer{}, fmt.Errorf("API of type '%s' requires a '%s', but none was defined", a, "scope")
-		}
-		if v := parsed.CustomValues["actionType"]; v == "" {
-			return pointer.DeletePointer{}, fmt.Errorf("API of type '%s' requires a '%s', but none was defined", a, "actionType")
-		}
-		if v := parsed.CustomValues["domain"]; v == "" {
-			return pointer.DeletePointer{}, fmt.Errorf("API of type '%s' requires a '%s', but none was defined", a, "domain")
-		}
+	return dp, nil
+}
 
-		return pointer.DeletePointer{
-			Type:       parsed.Type,
-			Identifier: parsed.ConfigName,
-			Scope:      parsed.Scope,
-			ActionType: parsed.CustomValues["actionType"],
-			Domain:     parsed.CustomValues["domain"],
-		}, nil
+func verifyAPIEntry(parsed persistence.DeleteEntry, a api.API) error {
+	if parsed.ConfigId != "" {
+		return errors.New("'id' is not supported for this API")
+	}
+	if parsed.Project != "" {
+		return errors.New("'project' is not supported for this API")
+	}
+	if parsed.ConfigName != "" && parsed.ObjectId != "" {
+		return errors.New("'name' and 'objectId' can't be used together to define an entry")
+	}
+	if parsed.ConfigName == "" && parsed.ObjectId == "" {
+		return errors.New("a 'name' or a 'originId' is required, but none was defined")
 	}
```

Comment:
We need to update the name for `originId` or `objectId` in the error messages here